File: 128.-Longest-Consecutive-Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        logest=0
        un_set=set(nums)
        if len(nums)==0:
            return 0
        for i in un_set:
            if  (i-1) not in  un_set:
                c=1
                x=i
                while (x+1) in  un_set:
                    x+=1
                    c+=1
                logest= max(logest,c)
        return logest

        

File: 128.-Longest-Consecutive-Sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if (nums.length===0){
        return 0
    }
    let long=0
    let un_set= new Set(nums)
    
    for (let i of un_set){
        if (! un_set.has(i-1) ){
            let x=i
            let c=1
            while (un_set.has(x+1) ){
                x++
                c++
            }
            long=Math.max(long,c)
        }
    }
    return long
    
};
